test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -0,0 +1,415 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests.Scale
+{
+    [Trait(TestTraits.CategoryTraitName, TestTraits.ScaleMonitoring)]
+    public class ScaleHostEndToEndTests
+    {
+        private const string Function1Name = "Function1";
+        private const string Function2Name = "Function2";
+        private const string Function3Name = "Function3";
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ScaleManager_GetScaleStatusAsync_ReturnsExpected(bool tbsEnabled)
+        {
+            var triggerMetadata = new List<TriggerMetadata>()
+            {
+                new TriggerMetadata(Function1Name, new JObject { { "type", "testExtensionATrigger" } }),
```

Comment:
[Mathew]: I'd like to discuss more why this `Properties` bag is needed for triggers, but for now assuming it is, I retain it in the API.

[Alexey]: This is needed to pass `AzureComponentFactory` for example (maybe some other properties in the future). We can not define `AzureComponentFactory` in `TriggerMetadata` as it would bring another nuget package reference and we do no want that.

Reply:
This is solved by IAzureComponentFactoryProvider in WebJobs.Host.Storage